debian/patches/0061-backport-scale-vt.patch

```diff
@@ -294,7 +294,7 @@ new file mode 100644
 +    outlink->hw_frames_ctx = av_hwframe_ctx_alloc(hw_frame_ctx_in->device_ref);
 +    hw_frame_ctx_out = (AVHWFramesContext *)outlink->hw_frames_ctx->data;
 +    hw_frame_ctx_out->format = AV_PIX_FMT_VIDEOTOOLBOX;
-+    hw_frame_ctx_out->sw_format = hw_frame_ctx_in->sw_format;
++    hw_frame_ctx_out->sw_format = s->colour_primaries == AVCOL_PRI_BT709 ? AV_PIX_FMT_NV12 : hw_frame_ctx_in->sw_format;
```

Comment:
It is better to expose a format option to make it consistent with other hardware `scale_*` filters. I don't think upstream likes this implicit format conversion either, unless it's documented somewhere that Apple explicitly requires this.

Reply:
In its current form, this is a quick fix for our downstream issue. We don't have many other options, as formats other than NV12 and P010 may make the encoder unhappy and causing frame ordering issue. However, I do agree that an option is needed for an upstream patch.

Reply:
Yeah it's a stopgap. Actually a potential breakage is that `tonemap_opencl` requires `p010le` input. This may fail when used with `scale_vt` since we can't set any format option.

Reply:
I noticed that, so I added this. When we are not using `scale_vt` to perform tone mapping, the output frame will retain the format as is. Therefore, the OpenCL tone mapping can still receive a P010 input. However, when we are setting color primaries, it means that we are using `scale_vt` itself to perform tone mapping, and its output is unlikely to be used with OpenCL tone mapping.

Reply:
> the output frame will retain the format as is. Therefore, the OpenCL tone mapping can still receive a P010 input

That's because we currently only have hardware decoding enabled for [4:2:0 sub-sampling](https://github.com/jellyfin/jellyfin/blob/d10ad6c383954e6acb76581d5459aa9740c66688/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L6105). It will absolutely fail when we enable 4:2:2 and 4:4:4. This is what we are going to do in JF 10.10/11, by adding a tool to detect the real hardware capabilities.